### PR TITLE
Sensitive map/list members

### PR DESCRIPTION
**Pull request by alextwoods**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

1. To make sure we include your contribution in the release notes, please make sure to add description entry for your changes in the "unreleased changes" section of the `CHANGELOG.md` file (at corresponding gem). For the description entry, please make sure it lives in one line and starts with `Feature` or `Issue` in the correct format.

2. For generated code changes, please checkout below instructions first:
  https://github.com/aws/aws-sdk-ruby/blob/version-3/CONTRIBUTING.md

Thank you for your contribution!

### CODE REVIEW

File: gems/aws-sdk-core/spec/aws/log/param_filter_spec.rb [hunk 1]
```diff
@@ -26,6 +26,13 @@ class OldServiceType < Struct.new(
         include Aws::Structure
       end
 
+      class ComplexSensitiveType < Struct.new(
```
Comment (mullermp) [inline gems/aws-sdk-core/spec/aws/log/param_filter_spec.rb:29]: We should handle document too and have a test for it.

File: gems/aws-sdk-core/spec/aws/log/param_filter_spec.rb [hunk 2]
```diff
@@ -55,6 +62,16 @@ class OldServiceType < Struct.new(
           expect(filtered).to eq(sensitive_member: '[FILTERED]')
         end
 
+        it 'filters nested sensitive params' do
```
Comment (mullermp) [inline gems/aws-sdk-core/spec/aws/log/param_filter_spec.rb:65]: Are map values that are primitive like strings handled? Can we test for that?

Comment (alextwoods) [reply gems/aws-sdk-core/spec/aws/log/param_filter_spec.rb:65]: This was actually an unrelated issue that I discovered while trying to make the code generation fix. 

And yes, maps and documents are handled, but this spec doesn't let us really test for that - we would need specs in the build tools.

File: build_tools/aws-sdk-code-generator/spec/interfaces/sensitive_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+describe 'Types Interface:' do
+  describe 'Sensitive members' do
+    before(:all) do
+      SpecHelper.generate_service(['Sensitive'], multiple_files: false)
+    end
+
+    let(:token) { 'token' }
+
+    let(:token_provider) { Aws::StaticTokenProvider.new(token) }
```
Comment (mullermp) [inline build_tools/aws-sdk-code-generator/spec/interfaces/sensitive_spec.rb]: I don't think these are needed

File: build_tools/aws-sdk-code-generator/spec/interfaces/sensitive_spec.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      )
+    end
+
+    describe '#kitchen_sink' do
+      it 'filters all sensitive members' do
+        resp = client.kitchen_sink_operation
+        puts resp
+        expect(resp.to_s).to eq(
+          (<<-OUTPUT
+{:blob=>"[FILTERED]", :boolean=>"[FILTERED]", :double=>"[FILTERED]", :empty_struct=>{}, :float=>"[FILTERED]", :integer=>"[FILTERED]", :json_value=>"[FILTERED]", :list_of_strings=>"[FILTERED]", :list_of_structs=>"[FILTERED]", :long=>"[FILTERED]", :map_of_lists_of_strings=>"[FILTERED]", :map_of_maps=>"[FILTERED]", :map_of_strings=>"[FILTERED]", :map_of_structs=>"[FILTERED]", :simple_struct=>"[FILTERED]", :string=>"[FILTERED]", :struct_with_json_name=>{:value=>"[FILTERED]"}, :timestamp=>"[FILTERED]"}
```
Comment (mullermp) [inline build_tools/aws-sdk-code-generator/spec/interfaces/sensitive_spec.rb]: Does it make sense to multiline this for readability or dynamically generate this as an expectation?

File: gems/aws-sdk-core/CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,8 @@
 Unreleased Changes
 ------------------
 
+* Issue - Ensure that nested sensitive members are filtered.
```
Comment (mullermp) [inline gems/aws-sdk-core/CHANGELOG.md]: It should be lists and maps too right?

**Review by mullermp — APPROVED** [review]

File: build_tools/aws-sdk-code-generator/spec/interfaces/sensitive_spec.rb (hunk 3 above)
Comment (alextwoods) [reply build_tools/aws-sdk-code-generator/spec/interfaces/sensitive_spec.rb]: removed.

File: build_tools/aws-sdk-code-generator/spec/interfaces/sensitive_spec.rb (hunk 4 above)
Comment (alextwoods) [reply build_tools/aws-sdk-code-generator/spec/interfaces/sensitive_spec.rb]: Yeah, I'll multiline it

File: gems/aws-sdk-core/CHANGELOG.md (hunk 5 above)
Comment (alextwoods) [reply gems/aws-sdk-core/CHANGELOG.md]: actually the fix in core only applies to nested sensitive members.  the codegen changes are what fixes lists and maps.

**Pull request merged**